public/docs/dart/latest/guide/dependency-injection.jade

```diff
@@ -1,6 +1,782 @@
+    creates a new `HeroListComponent`.
+<a id="di-metadata"></a>
+:marked
+  ### Creating the injector (implicitly)
+  When we introduced the idea of an injector above, we showed how to create
+  an injector and use it to create a new `Car`.
+
++makeExample('dependency-injection/dart/web/car/car-injector.dart','injector-create-and-call')(format=".")
+
+:marked
+  We won't find code like that in the Tour of Heroes or any of our other samples.
+  We *could* write code [like that](#explicit-injector) if we *had* to which we rarely do.
+  Angular takes care of creating and calling injectors
+  when it creates components for us whether through HTML markup, as in `<hero-list></hero-list>`,
+  or after navigating to a component with the [router](./router.html).
+  Let Angular do its job and we'll enjoy the benefits of automated dependency injection.
+
+  ### Singleton services
+  Dependencies are singletons within the scope of an injector.
+  In our example, there is a single `HeroService` instance shared among the
+  `HeroesComponent` and its `HeroListComponent` children.
+
+  However, Angular DI is an hierarchical injection
+  system which means nested injectors can create their own service instances.
+  Learn more about that in the [Hierarchical Injection](./hierarchical-dependency-injection.html) chapter.
+
+  ### Testing the component
+  We emphasized earlier that designing a class for dependency injection makes the class easier to test.
+  Listing dependencies as constructor parameters may be all we need to test application parts effectively.
+
+  For example, we can create a new `HeroListComponent` with a mock service that we can manipulate
+  under test:
+
++makeExample('dependency-injection/dart/web/test.component.dart', 'spec')(format='.')
+.l-sub-section
+  :marked
+    Learn more in the [Testing](../testing/index.html) chapter.
+
+:marked
+  ### When the service needs a service
+  Our `HeroService` is very simple. It doesn't have any dependencies of its own.
+
+
+  What if it had a dependency? What if it reported its activities through a logging service?
+  We'd apply the same *constructor injection* pattern,
+  adding a constructor that takes a `logger` parameter.
+
+  Here is the revision compared to the original.
+
++makeTabs(
+`dependency-injection/dart/web/heroes/hero.service.2.dart,
+dependency-injection/dart/web/heroes/hero.service.1.dart`,
+null,
+`app/heroes/hero.service (v.2),
+app/heroes/hero.service (v.1)`)
+
+:marked
+  The constructor now asks for an injected instance of a `Logger` and stores it in a private property called `_logger`.
+  We call that property within our `getHeroes()` method when anyone asks for heroes.
+
+  <a id="injectable"></a>
+  ### Why *@Injectable*?
+  Notice the `@Injectable()` decoration above the service class.
+  We haven't seen `@Injectable()` before.
+  As it happens, we could have added it to our first version`HeroService`.
+  We didn't bother because we didn't need it then.
```

Comment:
`@Injectable` is needed if you use `bootstrap` method. However if you use `Injector.resolveAndCreate` the annotation `@Injectable` is not needed.